acados/ocp_nlp/ocp_nlp_constraints_bghp.c

```diff
@@ -1073,12 +1088,43 @@ void ocp_nlp_constraints_bghp_update_qp_matrices(void *config_, void *dims_, voi
         model->p->evaluate(model->p, ext_fun_type_in, ext_fun_in, ext_fun_type_out, ext_fun_out);
 
         // SCQP Hessian
-        double lam = blasfeo_dvecex1(memory->lam, 2 * (nb + ng) + nh) -
-                     blasfeo_dvecex1(memory->lam, nb + ng);
-
-        blasfeo_dsyrk_ln(nx + nu, np, 2 * lam, &work->jacobian_quadratic, 0, 0,
-                         &work->jacobian_quadratic, 0, 0, 1.0, memory->RSQrq, 0, 0, memory->RSQrq,
-                         0, 0);
+        
+        for (int i = 0; i < nh; i++) { 
+            // TODO(andrea): @giaf: if the lower bound is active we shouldn't consider its contribution, right?
+            // Removing this.
+            // double lam_i = blasfeo_dvecex1(memory->lam, 2 * (nb + ng) + nh) -
+            //              blasfeo_dvecex1(memory->lam, nb + ng);
+
+            double lam_i = blasfeo_dvecex1(memory->lam, 2 * (nb + ng) + nh + i);
```

Comment:
add a note stating that only the upper constraint is considered in the hessian approximation computation, since an active lower constraint would add a negative contribution to the hessian and (possibly) result in an indefinite hessian